packages/opc-feedback/src/opc-feedback.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2021 Red Hat One Platform
+ *
+ * MIT
+ *
+ * opc-feedback.ts enables the user to collect the feedback from the user
+ *
+ * @summary opc-feedback.ts will set the properties 
+ * @author Rigin Oommen
+ *
+ * Created at     : 2021-01-18 14:53:24 
+ * Last modified  : 2021-01-25 21:43:54
+ */
+
+import { LitElement, html, property, customElement, internalProperty, query } from 'lit-element';
+import style from './opc-feedback.scss';
+import { repeat } from 'lit-html/directives/repeat.js';
+
+@customElement('opc-feedback')
+export class OpcFeedback extends LitElement {
+  @property({ type: String, attribute: 'url' }) url = '/feedback';
+  @internalProperty() openFeedbackModal = false;
+  @internalProperty() openConfirmationModal = false;
+  @internalProperty() protected summary = '';
+  @internalProperty() experience = '';
+  @internalProperty() category = '';
+  @internalProperty() experienceList = [{
+    name: 'Excellent',
+    assetUrl: './assets/happy.svg'
+  },
+  {
+    name: 'Good',
+    assetUrl: './assets/good.svg'
+  },
+  {
+    name: 'Sad',
+    assetUrl: './assets/sad.svg'
+  }];
+  @internalProperty() error = '';
+  @internalProperty() errorList = [{
+    name: 'Slow Loading',
+  },
+  {
+    name: 'App Crashed',
+  },
+  {
+    name: 'Navigation',
+  },
+  {
+    name: 'Not Responsive',
+  },
+  {
+    name: 'Other',
+  }];
+
+  @query('textarea') textarea: HTMLTextAreaElement;
+  
+  static get styles() {
+    return [style];
+  }
+
+  resetForm() {
+    this.summary = '';
+    this.experience = '';
+    this.category = '';
+    this.error = '';
+    this.textarea.value = '';
+  }
+
+  submitFeedback() {
+    this.category = (this.experience !== 'Sad') ? 'FEEDBACK' : 'BUG';
+    this.dispatchEvent(new CustomEvent('opc-feedback:submit', {
+      detail: {
+        message: 'Submitted the Feedback',
+        data: {
+          summary: this.summary,
+          experience: this.experience,
+          category: this.category,
+          stackInfo: {
+            "stack": navigator.appVersion,
+            "path": window.location.pathname,
+            "error": (this.error) ? `${this.error}` : ''
+          }
+        }
+      }
+    }));
+    this.resetForm();
+  }
+
+  toggle() {
+    this.openFeedbackModal = !this.openFeedbackModal;
+  }
+  
+  render() {
+    return html`
+    <link type="text/css" rel="stylesheet" href="https://unpkg.com/@patternfly/patternfly/patternfly.css"
+      crossorigin="anonymous" />
+    <link type="text/css" rel="stylesheet" href="https://unpkg.com/@patternfly/patternfly/patternfly-addons.css"
+      crossorigin="anonymous" />
+    <dialog class="op-feedback__panel" .open="${this.openFeedbackModal}">
+      <form class="feedback-form" id="feedbackForm">
+        <header>
+          <h3 class="pf-u-font-weight-normal pf-u-font-size-lg pf-u-text-align-center pf-u-m-0">How was your overall experience?.</h3>
```

Comment:
these questions should be dynamic

Reply:
Implemented default values and provision to override the default tempate values